tests/test_variable.py

```diff
@@ -137,5 +140,30 @@ def test_invalid_value_type(self):
         with self.assertRaises(AssertionError):
             chainer.Variable(1)
 
+    def test_grad_type_check_pass(self):
+        a = chainer.Variable(np.empty((3,), dtype=np.float32))
+        a.grad = np.ndarray((3,), dtype=np.float32)
```

Comment:
AFAIK, there is no such function that checks if the code raises nothing. When the code raises an exception without catch, the test is failed with the stack trace as same as usual assertion failures.